### ISSUE TITLE
Conforming Alternate Version (CAV) and the mechanism to reach/achieve it

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#5373]
This issue tries to summarise the lengthy discussion (with additional tangents) in [Are there circumstances in which an overlay tool could be seen as providing a conforming alternate version for a non-conforming website? #2750](https://github.com/w3c/wcag/issues/2750), getting to the crux of the argument presented by @detlevfisher

### Background

Note that this is *not* about "can overlay tools ever remediate a broken/inaccessible site to the point of making it compliant". This aspect would then be determined by evaluating the "end result" after the overlay made its changes ... if the resulting version of the site is inaccessible, then clearly it would fail and not count as a conforming alternate version (CAV) https://www.w3.org/TR/WCAG22/#dfn-conforming-alternate-versions.

However, Detlev's point is more specifically about the *mechanism* to reach a conforming alternate version.

* the initial answer in #2750 (which was agreed by AGWG, but was then reopened) was that yes, if an overlay does lead to a conforming version, the overlay counts as a mechanism to reach the CAV - as it's functionally equivalent to Note 7 which already allows "user preferences" as a way to produce a CAV.
* Detlev's counterargument was that an overlay can't be counted as a valid mechanism, because the mechanism to reach a CAV must be accessibility-supported https://www.w3.org/TR/WCAG22/#dfn-accessibility-supported - that it must satisfy all WCAG criteria by itself. Detlev's argument is that since most/all overlays don't have either a single control that is clearly labelled as being the one to enable the CAV, and there are no specific instructions that explain to a visitor which options they need to enable to reach the CAV, that this fails 2.4.6 Headings and Labels (or even 3.3.2 Labels or Instructions)

The counter to the counterargument in the discussion is that not having an explicit label/instruction that indicates "this enables the accessible/conforming version" is not a failure of 2.4.6/3.3.2. Provided that the controls in the overlay have their own specific labels (e.g. "Increase contrast", "Underline all links", or similar), there's no additional normative requirement that you MUST include this extra label/instruction, or else it doesn't count as accessibility supported.

---

For reference, here are some common examples of what accessibility overlays in the wild look like. These often include a myriad of options (most of them not related to remediating accessibility per se, but providing customisation).

> ![composite view of an overlay 'Accessibility' dialog, with ready-made profiles, settings for text, colours, reading and focus - 19 individual options overall](https://github.com/user-attachments/assets/85c49789-46b6-4fa6-bf51-b8708c127a18)

> ![an 'Accessibility options' overaly, with settings for text, color and contrast, tools - 16 individual options overall](https://github.com/user-attachments/assets/df3f37e3-97c1-487d-929d-0620de97cb0a)

> ![an 'Accessibility Menu' overlay, with profiles, content adjustments for text, links, letter spacing, line height, font weight, color adjustments, custom color section options, and tools - approximately 20 options overall](https://github.com/user-attachments/assets/092d8b08-5dfc-472e-be23-6fee8aba20e9)

> ![an 'Accessibility Adjustments' overlay, showing different profiles - e.g. seizure safe profile - and content adjustments](https://github.com/user-attachments/assets/9f550fb8-6ce3-4e91-a5b7-eabfe6e87c33)

> ![an 'Accessibility Menu' overlay, with profiles and various options for smart contrast, pausing animations, increasing contrast, highlighting links, etc](https://github.com/user-attachments/assets/9752be4f-8278-4b37-9601-b24c54b1e074)

---

Also for reference, here's an example of accessibility settings in a site's "user preferences". Again, note how these just "say what they do", and don't include additional labels/instructions that explain which ones need to be enabled to reach a CAV.

> ![GitHub's profile 'Accessibility' section, with highlighted checkboxes for 'Character keys' and radio buttons to enable/disable link underlines](https://github.com/user-attachments/assets/cd585271-c4ee-44e1-b647-81e7f87be51c)

---

#### Question

For a simplified argument, assume that the site, by default, has low contrast (fails 1.4.3 Contrast (Minimum) and 1.4.11 Non-text Contrast). But there is a floating basic overlay that provides a control "Contrast" in the top right corner of the viewport - once that is activated, all 1.4.3 and 1.4.11 failures disappear.

> ![a site header, all blue with some decorative lines/patterns, with a single 'Contrast' button in the corner](https://github.com/user-attachments/assets/9ef837ed-8228-491c-8e00-d918c68982ec)

Note that we are also taking as a given that the mechanism itself in this case, even if the site by default has low contrast, must itself have sufficient contrast (so that users who struggle with low contrast can actually find it in the first place), that it can be reached and activated with the keyboard, that the default page still doesn't include things that would violate non-interference https://www.w3.org/TR/WCAG22/#cc5

Does the result, after activating the "Contrast" control:

1) count as a conforming alternate version (CAV), and the overlay/button was the mechanism that allowed the user to reach the CAV? or
2) since the overlay didn't have a clear label or instruction that says something along the lines of "enable the 'Contrast' control to reach the conforming alternate version" or words to that effect, the overlay fails 2.4.6 (or 3.3.2), and is therefore not accessibility supported, so can't count as a valid mechanism to reach a CAV

**Comment by patrickhlauke** [comment]
(temporarily locking this issue, as we refine this to accurately reflect the question from @detlevhfischer - it will be unlocked when the issue states the underlying question correctly, and the issue is sent to the wider AGWG for discussion)
